### PR TITLE
[Merged by Bors] - TY 2374 Rename DocumentFeedback to UserReaction

### PR DESCRIPTION
**Pull request by acrrd**
Rename the dart class to have the same name we have in rust

### CODE REVIEW

**Review by rustonaut — APPROVED** [review]
lgtm

File: discovery_engine/lib/src/api/events/client_events.dart [hunk 1]
```diff
@@ -102,7 +102,7 @@ class ClientEvent with _$ClientEvent {
   @Implements<DocumentClientEvent>()
   const factory ClientEvent.documentFeedbackChanged(
     DocumentId documentId,
-    DocumentFeedback feedback,
+    UserReaction feedback,
```
Comment (Robert-Steiner) [inline discovery_engine/lib/src/api/events/client_events.dart:105]: what about also changing `feedback` to `reaction`?

**Comment by acrrd** [comment]
Renamed more consistently. Please take another look.

File: discovery_engine/lib/src/discovery_engine_base.dart [hunk 2]
```diff
@@ -255,23 +255,23 @@ class DiscoveryEngine {
...
+  /// [UserReaction] variants are defined as:
+  /// - [UserReaction.positive] indicates that the [Document] was **liked**
+  /// - [UserReaction.negative] indicates that the [Document] was **diliked**
+  /// - [UserReaction.neutral] as a default **neutral** state of the [Document].
   ///
   /// In response it can return:
   /// - [ClientEventSucceeded] indicating a successful operation
   /// - [EngineExceptionReason] indicating a failed operation, with a reason
   /// for such failure.
   Future<EngineEvent> changeDocumentFeedback({
```
Comment (Robert-Steiner) [inline discovery_engine/lib/src/discovery_engine_base.dart:269]:
```suggestion
  Future<EngineEvent> changeUserReaction({
```

File: discovery_engine/lib/src/domain/document_manager.dart [hunk 3]
```diff
@@ -40,24 +40,25 @@ class DocumentManager {
   /// Fails if the event [evt] does not have a handler implemented.
   Future<void> handleDocumentClientEvent(DocumentClientEvent evt) =>
       evt.maybeWhen(
-        documentFeedbackChanged: (id, fdbk) => updateDocumentFeedback(id, fdbk),
+        userReactionChanged: (id, reaction) =>
+            updateDocumentFeedback(id, reaction),
```
Comment (Robert-Steiner) [inline discovery_engine/lib/src/domain/document_manager.dart:44]:
```suggestion
            updateUserReaction(id, reaction),
```

File: discovery_engine/lib/src/domain/document_manager.dart [hunk 4]
```diff
@@ -40,24 +40,25 @@ class DocumentManager {
...
         documentTimeSpent: (id, mode, sec) =>
             addActiveDocumentTime(id, mode, sec),
         orElse: throw UnimplementedError('handler not implemented for $evt'),
       );
 
-  /// Update feedback for the given document.
+  /// Update user reaction for the given document.
   ///
   /// Fails if [id] does not identify an active document.
   Future<void> updateDocumentFeedback(
```
Comment (Robert-Steiner) [inline discovery_engine/lib/src/domain/document_manager.dart:53]:
```suggestion
  Future<void> updateUserReaction(
```

File: discovery_engine/lib/src/domain/engine/engine.dart [hunk 5]
```diff
@@ -31,7 +31,7 @@ abstract class Engine {
     DocumentId docId, {
     required Uint8List smbertEmbedding,
     required Duration seconds,
-    required DocumentFeedback reaction,
+    required UserReaction reaction,
   });
 
   /// Process the feedback about the user reacting to a document.
```
Comment (Robert-Steiner) [inline discovery_engine/lib/src/domain/engine/engine.dart:37]:
```suggestion
  /// Process the user's reacting to a document.
```

File: discovery_engine/test/discovery_engine/change_document_feedback_test.dart [hunk 6]
```diff
@@ -41,7 +41,7 @@ void main() {
       expect(
         engine.changeDocumentFeedback(
```
Comment (Robert-Steiner) [inline discovery_engine/test/discovery_engine/change_document_feedback_test.dart:42]:
```suggestion
        engine.changeUserReaction(
```

File: discovery_engine/test/discovery_engine/change_document_feedback_test.dart [hunk 7]
```diff
@@ -53,7 +53,7 @@ void main() {
       final engine = await createEngineWithEntryPoint(withErrorResponse);
       final response = await engine.changeDocumentFeedback(
```
Comment (Robert-Steiner) [inline discovery_engine/test/discovery_engine/change_document_feedback_test.dart:54]:
```suggestion
      final response = await engine.changeUserReaction(
```

File: discovery_engine/test/discovery_engine/change_document_feedback_test.dart [hunk 8]
```diff
@@ -70,7 +70,7 @@ void main() {
       final engine = await createEngineWithEntryPoint(withWrongEventResponse);
       final response = await engine.changeDocumentFeedback(
```
Comment (Robert-Steiner) [inline discovery_engine/test/discovery_engine/change_document_feedback_test.dart:71]:
```suggestion
      final response = await engine.changeUserReaction(
```

File: discovery_engine/test/document_manager_test.dart [hunk 9]
```diff
@@ -95,22 +95,22 @@ Future<void> main() async {
...
-      doc2.feedback = DocumentFeedback.neutral;
+      doc2.userReaction = UserReaction.neutral;
       data.viewTime.clear();
     });
 
-    test('update absent document feedback', () async {
+    test('update absent document user reaction', () async {
       expect(
-        () => mgr.updateDocumentFeedback(id3, DocumentFeedback.positive),
+        () => mgr.updateDocumentFeedback(id3, UserReaction.positive),
```
Comment (Robert-Steiner) [inline discovery_engine/test/document_manager_test.dart:106]:
```suggestion
        () => mgr.updateUserReaction(id3, UserReaction.positive),
```

same below

**Review by Robert-Steiner — COMMENTED** [review]
There are still some more methods/events/doc strings that have `DocumentFeedback` in their names:

`changeDocumentFeedback`
`updateDocumentFeedback`
`DocumentFeedbackChanged`

we should also update `docs/events.md` and `docs/models.md`

**Comment by acrrd** [comment]
Thanks, I was searching for `feedback` but I forgot it was case insensitive.

File: discovery_engine/lib/src/domain/engine/engine.dart [hunk 10]
```diff
@@ -31,15 +31,15 @@ abstract class Engine {
     DocumentId docId, {
     required Uint8List smbertEmbedding,
     required Duration seconds,
-    required DocumentFeedback reaction,
+    required UserReaction reaction,
   });
 
-  /// Process the feedback about the user reacting to a document.
+  /// Process the user's reacting to a document.
```
Comment (Robert-Steiner) [inline discovery_engine/lib/src/domain/engine/engine.dart:37]: my bad 
```suggestion
  /// Process the user's reaction to a document.
```

**Review by Robert-Steiner — COMMENTED** [review]
looks good but I think it would be good to also update the documentation in the `docs` folder

**Comment by acrrd** [comment]
The doc is become more a duplicate of the code then an integration of it.

**Comment by acrrd** [comment]
bors merge

**Pull request closed without merging**